src/nodemetadata.cpp

```diff
@@ -77,11 +90,38 @@ bool NodeMetadata::empty() const
 	return Metadata::empty() && m_inventory->getLists().size() == 0;
 }
 
+bool NodeMetadata::getPrivate(const std::string &name) const
+{
+	return m_privatevars.count(name) != 0;
+}
+
+void NodeMetadata::setPrivate(const std::string &name, bool set)
+{
+	if (set)
+		m_privatevars.insert(name);
+	else
+		m_privatevars.erase(name);
+}
+
+int NodeMetadata::countNonPrivate() const
+{
+	// m_privatevars can contain names not actually present
+	// DON'T: return m_stringvars.size() - m_privatevars.size();
+	int n = 0;
+	for (StringMap::const_iterator
+			it = m_stringvars.begin();
+			it != m_stringvars.end(); ++it) {
+		if (getPrivate(it->first) == false)
```

Comment:
in C++11 it will be nicer using std::find(m_stringvars.begin, end, [] { return getPrivate(it->first); }